### PR TITLE
Enable IPC module tests

### PR DESCRIPTION
**Pull request by araujol**
This commit enables the tests for the IPC ZeroMQ module.

Signed-off-by: Luis Araujo <luis.araujo@collabora.co.uk>

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -78,14 +75,20 @@ def run_vsm(self, name, input_data, expected_output, use_initial=True):
...
+            # they will be stuck on 'receive'.
+            if send_quit:
                 self._send('quit', '')
 
-            output = self._receive()
+            sig, val = self._receive()
             process.terminate()
 
-            self.assertEqual(output, format_ipc_output(expected_output))
+            # Read state dump from log file vsm.log
```
Comment (treitter) [inline tests.py]: cut off the trailing " vsm.log" in this comment so you don't alias the content of VSM_LOG_FILE

File: tests.py [hunk 2]
```diff
@@ -11,24 +11,18 @@
 
 
 RULES_PATH = os.path.abspath(os.path.join('.', 'sample_rules'))
+SIGNAL_FORMAT = '{},[SIGNUM],\'{}\'\n'
+VSM_LOG_FILE = 'vsm.log'
```
Comment (treitter) [inline tests.py]: I think we should use a non-default file (like vsm-tests.log) so we don't potentially overwrite an existing file from a previous (manual) run.

File: tests.py [hunk 3]
```diff
@@ -54,15 +48,18 @@ def _receive(self):
...
         conf = os.path.join(RULES_PATH, name + '.yaml')
         initial_state = os.path.join(RULES_PATH, name + '.initial.yaml')
 
         cmd = ['./vsm' ]
 
-        # direct verbose output (including state dumps) to stdout so the tests
-        # can parse them
-        cmd += ['--log-file=-']
+        # Read from the log file for the zeromq tests.
+        if TestVSM.ipc_module != 'zeromq':
```
Comment (treitter) [inline tests.py]: Assuming we're using a non-default log file in `VSM_LOG_FILE` above, specify it here in case `TestVSM.ipc_module == 'zeromq'`

**Review by treitter — CHANGES_REQUESTED** [review]
Generally looks good; I'd just like a few minor changes.

**Comment by araujol** [comment]
Ok, I applied the suggested changes.

I also changed **self.ipc_module** to **TestVSM.ipc_module** , since it can show better it is a class variable.

**Pull request merged**